compiler/rustc_passes/src/check_attr.rs

```diff
@@ -2286,6 +2286,11 @@ impl<'tcx> Visitor<'tcx> for CheckAttrVisitor<'tcx> {
         };
 
         self.check_attributes(expr.hir_id, expr.span, target, None);
+        if let hir::ExprKind::Struct(_, fields, _) = expr.kind {
+            for field in fields {
+                self.check_attributes(field.hir_id, field.span, Target::PatField, None);
```

Comment:
Do expression fields behave exactly like `PatField`?  Should the `PatField`'s name reflect that?